boards/arm/stm32f746g_disco/stm32f746g_disco.dts

```diff
@@ -119,3 +120,36 @@
 		     &eth_txd0_pg13
 		     &eth_txd1_pg14>;
 };
+
+&quadspi {
+	pinctrl-0 = <&quadspi_clk_pb2 &quadspi_bk1_ncs_pb6
+		     &quadspi_bk1_io0_pd11 &quadspi_bk1_io1_pd12
+		     &quadspi_bk1_io2_pe2 &quadspi_bk1_io3_pd13>;
+	status = "okay";
+
+	n25q128a1: qspi-nor-flash@0 {
+		compatible = "st,stm32-qspi-nor";
+		label = "N25Q128A1";
+		reg = <0>;
+		qspi-max-frequency = <108000000>;
+		/* 128 Megabits = 16 Megabytes */
+		size = <0x8000000>;
+		status = "okay";
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			slot1_partition: partition@0 {
+				label = "image-1";
+				reg = <0x00000000 0x000D8000>;
+				};
+
+			storage_partition: partition@D8000 {
+				label = "storage";
+				reg = <0x000D8000 DT_SIZE_M(7)>;
```

Comment:
Not sure which values I should put for the partition sizes. I just copied these from _disco_l475_iot1_. Comments appreciated.

Reply:
This board and disco have the same internal flash, so you can use the same image flash, but for storage, you can use the max amount minus image size (around 15M).
Though, to be coherent, I'd suggest adding the flash partition on internal flash as well.

Reply:
Actually, had to resize flash image because sector size is much larger for F7, so we need at least 256KiB scratch partition.